src/components/DateTimePicker/TimePicker.tsx

```diff
@@ -52,22 +58,93 @@ export class TimePicker extends React.Component<ITimePickerProps, void> {
         this.props.onTimeChanged(this.props.hour, value);
     }
 
+    normalizeHours(hour: number, is24Hour: boolean, isPeriodPm: boolean) {
+        if (!is24Hour) {
+            if (hour > 12) {
+                hour = 12;
+            }
+            if (isPeriodPm) {
+                if (hour < 12) {
+                    hour += 12;
+                }
+            } else {
+                if (hour === 12) {
+                    hour = 0;

```

Comment:
If a displayed time is 12 AM, it's 0 in 24-hour format and needs to be set as 0 in outside Date representation (props.hour).